src/components/tenant-selector.tsx

```diff
@@ -0,0 +1,83 @@
+"use client";
+
+import { useState } from "react";
+
+import { Button } from "@/components/ui/button";
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+import { Label } from "@/components/ui/label";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+
+interface UserInfo {
+  access: { tenant?: { domain: string } }[];
+}
+
+interface TenantSelectorProps {
+  userInfo: UserInfo;
+  handleSetTenant: (tenant: string) => void;
+  handleLogout: () => void;
+}
+
+export default function TenantSelector({
+  userInfo,
+  handleSetTenant,
+  handleLogout,
+}: TenantSelectorProps) {
+  const [isLoading, setIsLoading] = useState(false);
+
+  return (
+    <Card className="w-full max-w-md">
+      <CardHeader>
+        <CardTitle>Select Tenant</CardTitle>
+        <CardDescription>
+          Select the tenant for use with the application
+        </CardDescription>
+      </CardHeader>
+      <CardContent>
+        <div className="grid w-full items-center gap-4">
+          <div className="flex flex-col space-y-1.5">
+            <Label htmlFor="tenant">Tenant</Label>
+            <Select
+              name="tenant"
+              onValueChange={async (tenant) => {
+                setIsLoading(true); // Disable logout when selecting tenant
```

Comment:
```suggestion
setIsLoading(true);
try {
  await handleSetTenant(tenant);
} finally {
  setIsLoading(false);
}
```

We should not leave the button in the loading state if the set tenant operation fails.

Reply:
make sense, i have updated